src/Plugin/Commerce/PaymentGateway/Ecp.php

```diff
@@ -50,12 +56,14 @@ public function createPayment(PaymentInterface $payment, $capture = TRUE) {
       );
       $result = $client->create($data);
       $remote_id = $result->id;
+      $merchant_transaction_id = $data['merchantTransactionId'];
     }
 
     // Mark the payment as pending; it will be marked as completed when we
     // receive the IPN signaling the payment was captured successfully.
     $payment->setState('pending');
     $payment->setRemoteId($remote_id);
+    $payment->set('bluesnap_merchant_transaction_id', $merchant_transaction_id);
```

Comment:
Instead of a new merchant_transaction_id field, couldn't we generate a UUID as we're doing now and manually set that as the ID of the payment entity and then just pass that as the merchant transaction ID? In Drupal it seems we can do that now:

https://www.drupal.org/project/drupal/issues/2885469

Reply:
@shabananavas 

`SHOW CREATE TABLE commerce_payment`
`payment_id` int(10) unsigned NOT NULL AUTO_INCREMENT`

The ID is an integer and it is being auto-incremented by the database. The issue that you reference talks about providing the ability for custom entities. It doesn't apply to all content entity type but to entity types where the decision has been made from the beginning.

Not possible.